empress/support_files/js/empress.js

```diff
@@ -1210,5 +1210,97 @@ define([
         this._events.selectedNodeMenu.hiddenCallback = callback;
     };
 
+    /**
+     * Calculate the number of samples in which a leaf node appears for each
+     * level of a given metadata field.
+     *
+     * @param {String} nodeName Name of the (leaf) node for which to calculate
+     *                          sample presence.
+     * @param {String} fields Metadata fields for which to calculate leaf
+     *                       sample presence.
+     * @return {Object} ctData returns a mapping of field categories to number of
+     *                  samples in which the given leaf node is present and
+     *                  sample field value equals category.
+     */
+    Empress.prototype.computeLeafSamplePresence = function (nodeName, fields) {
+        var ctData = {};
+
+        for (var f = 0; f < fields.length; f++) {
+            var field = fields[f];
+            var obs = this._biom.getObsCountsBy(field, nodeName);
+            var categories = _.keys(obs);
+            ctData[field] = {};
+            for (var c = 0; c < categories.length; c++) {
+                var cat = categories[c];
+                ctData[field][cat] = obs[cat];
+            }
```

Comment:
I didn't notice this until now (this is probably my bad), but I think this code could be simplified a lot.

```suggestion
            var field = fields[f];
            ctData[field] = this._biom.getObsCountsBy(field, nodeName);
```